### PR TITLE
2226 detect boot state

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of https://github.com/kairos-io/kairos/issues/2226

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ go 1.20
 
 // This versions require go1.20
 replace (
+	github.com/kairos-io/kairos-sdk => /home/dimitris/workspace/kairos/kairos-sdk
```
Comment (jimmykarily) [inline go.mod:7]: Points to this: https://github.com/kairos-io/kairos-sdk/pull/69/files

File: pkg/uki/upgrade.go [hunk 2]
```diff
@@ -39,61 +36,41 @@ func (i *UpgradeAction) Run() (err error) {
 		return err
 	}
 	cleanup.Push(umount)
-	// TODO: Check number of existing UKI files
-	efiFiles, err := i.getEfiFiles()
+
+	// TODO: Get the size of the efi partition and decide if the images can fit
+	// before trying the upgrade.
+	// If we decide to first copy and then rotate, we need ~4 times the size of
+	// the artifact set [TBD]
```
Comment (jimmykarily) [inline pkg/uki/upgrade.go:43]: https://github.com/kairos-io/kairos/issues/2267

File: pkg/utils/common.go [hunk 3]
```diff
@@ -528,31 +528,3 @@ func UkiBootMode() state.Boot {
 	}
 	return state.Unknown
 }
-
-// SystemdBootConfReader reads a systemd-boot conf file and returns a map with the key/value pairs
-func SystemdBootConfReader(filePath string) (map[string]string, error) {
```
Comment (jimmykarily) [inline pkg/utils/common.go:533]: Moved [here](https://github.com/kairos-io/kairos-sdk/pull/69/files#diff-0df82f8689d3e8d881be0653ab4e9f8221c5835ec95a14930d11669e9c7f07efR358)

File: pkg/uki/install.go [hunk 4]
```diff
@@ -73,6 +74,9 @@ func (i *InstallAction) Run() (err error) {
 		return err
 	}
 
+	// TODO: Check if the size of the files we are going to copy, will fit in the
+	// partition. If not stop here.
```
Comment (jimmykarily) [inline pkg/uki/install.go:78]: https://github.com/kairos-io/kairos/issues/2267

**Review by mauromorales — APPROVED** [review]

**Pull request merged**